### ISSUE TITLE
The Bulk RESTful API should also use JSON format

### ISSUE THREAD

**Issue by cooniur** [elasticsearch/elasticsearch#9573]
The current bulk REST API is really, really, really hard to use, especially if one misses the last "\n" accidentally (anybody feels the same?), since it is not JSON format but plain text, it's difficult to check the syntax as apposed to JSON.

So, instead of the current plain text format:

    $ cat requests
    { "index" : { "_index" : "test", "_type" : "type1", "_id" : "1" } }
    { "field1" : "value1" }
    { "delete" : { "_index" : "test", "_type" : "type1", "_id" : "2" } }
    $ curl -s -XPOST localhost:9200/_bulk --data-binary @requests

why not using the following simple JSON format?

    $ curl -XPOST localhost:9200/_bulk -d '[
    {
      "action": "index",
      "params": { "_index" : "test", "_type" : "type1", "_id" : "1" },
      "data": { "field1": "value1" }
    },
    {
      "action": "delete",
      "params": { "_index" : "test", "_type" : "type1", "_id" : "2" }
    ]'

If the above is too wordy, how about the following?

    $ curl -XPOST localhost:9200/_bulk -d '[
      { "index" : { "_index" : "test", "_type" : "type1", "_id" : "1" } },
      { "field1" : "value1" },
      { "delete" : { "_index" : "test", "_type" : "type1", "_id" : "2" } }
    ]'

This would make programmers' life much much easier.

**Comment by dadoonet** [comment]
Please ask questions on the mailing list. Better place for this.

That said, have a look at http://www.elasticsearch.org/guide/en/elasticsearch/guide/current/distrib-multi-doc.html#bulk-format
